graylog2-server/src/main/java/org/graylog2/rest/RestTools.java

```diff
@@ -142,6 +141,10 @@ public static String getPathFromResource(Resource resource) {
         }
 
         return path;
+    }
 
+    public static Response.ResponseBuilder respondWithFile(String filename, Object entity) {
+        return Response.ok(entity)
+                .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=" + filename);
```

Comment:
How about adding overloaded methods that take the `Content-Length` and media type to be able to use the helper for cases like this?
https://github.com/Graylog2/graylog2-server/blob/1998a3d3f46e4380295333693fdef86828dbacc1/graylog2-server/src/main/java/org/graylog2/rest/resources/system/debug/bundle/SupportBundleResource.java#L89-L92

Reply:
I was also thinking about it, but had the feeling that it would not provide much value to the called. By returning the response builder it offers enough flexibility and using something like `respondWithFile(...).type(MediaType....)` or `respondWithFile(...).header(HttpHeaders.CONTENT_LENGTH, size)` it does not offer much room for errors, in contrast to the magic string of the content disposition header.

Reply:
Alright, I am convinced. Changed it in the PR :)